### PR TITLE
move validation function from ubuntu.py to validate.py

### PR DESCRIPTION
**Pull request by kaushik-rohit**
Changes proposed in this pull request:
- move validation function inside ubuntu.py to lib/validate.py

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Notify reviewers
@rohangoel96

### CODE REVIEW

File: lib/validate.py [hunk 1]
```diff
@@ -31,4 +36,119 @@ def errorMessage(value_number, expected_range, actual_value, fileName):
 
     """
     print "[Unexpected Value] of Arg", value_number, " @", fileName, "| EXPECTED_RANGE:", \
-        expected_range, "| GOT:", actual_value
\ No newline at end of file
+        expected_range, "| GOT:", actual_value
+        
+        
+# Correlational : statistical distribution as illustrated by box plot for degree
+def box_plot_for_degree(log_directory, output_directory, channel_name):
```
Comment (prasadtalasila) [inline lib/validate.py:43]: It's better to make a list parameter for channel names and iterste over the list. Right now, the list is static.

File: lib/validate.py [hunk 2]
```diff
@@ -31,4 +36,119 @@ def errorMessage(value_number, expected_range, actual_value, fileName):
...
+    cutoff = 0
+    for channel_name_iter in [["#kubuntu-devel"], ["#ubuntu-devel"], ["#kubuntu"]]:
```
Comment (prasadtalasila) [inline lib/validate.py]: Please change the list from static, fixed one to the list coming in via function parameter.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@kaushik-rohit please add the unit tests for validate.py.

**Comment by prasadtalasila** [comment]
@kaushik-rohit This being a light PR, it might br better to put in the modifications required for separaring the unit and functional tests in travis.

**Comment by kaushik-rohit** [comment]
@prasadtalasila fixed. With unit-test and functional test having separate coverage run, out code coverage will decrease (60%) also there are certain codeclimate issues due to moving of code from ubuntu.py (not monitored file) to validate.py. We can take this up in refactoring phase.

**Comment by prasadtalasila** [comment]
@kaushik-rohit The decrease in coverage is expected. There is a problem in travis.yml. The coverage report is not being uploaded to codeclimate. You can check the end of the log. codeclimate test reporter fails silently.

**Comment by kaushik-rohit** [comment]
@prasadtalasila That is not an error the token ($CODECLIMATE_REPO_TOKEN) is not exposed in pull requests. Once the pull request is merged it should worked correctly.

**Pull request merged**